app/gui/src/project-view/components/MarkdownEditor/TableEditorCell.vue

```diff
@@ -0,0 +1,124 @@
+<script setup lang="ts">
+import EditorRoot from '@/components/codemirror/EditorRoot.vue'
+import { highlightStyle } from '@/components/MarkdownEditor/highlight'
+import { ensoMarkdown } from '@/components/MarkdownEditor/markdown'
+import VueComponentHost, { VueHost } from '@/components/VueComponentHost.vue'
+import { EditorState } from '@codemirror/state'
+import { EditorView } from '@codemirror/view'
+import { minimalSetup } from 'codemirror'
+import { type ComponentInstance, onMounted, ref, toRef, useCssModule, watch } from 'vue'
+
+const editorRoot = ref<ComponentInstance<typeof EditorRoot>>()
+
+const props = defineProps<{
+  source: string
+}>()
+
+const vueHost = ref<ComponentInstance<typeof VueComponentHost>>()
+
+const editorView = new EditorView()
+const constantExtensions = [minimalSetup, highlightStyle(useCssModule()), EditorView.lineWrapping]
+
+watch([vueHost, toRef(props, 'source')], ([vueHost, source]) => {
+  if (!vueHost) return
```

Comment:
Props destructuring is actually a rather simplistic macro--it would be equivalent to replacing references to `source` with `props.source`. So the destructured bindings are reactive *values*, but they aren't reactive references that can be watched.

Reply:
Right, I didn't read the documentation to the end...